### ISSUE TITLE
Some font configurations make zed completely unresponsive.

### ISSUE THREAD

**Issue by shot-codes** [zed-industries/community#246]
**Describe the bug**
I set the buffer_font_family to "Iosevka" and Zed becomes completely unresponsive. I get a Mac spinning beach ball cursor, and when I force quit the app I am not able to open it again. I am only able to make it responsive again by changing the font configuration in the settings files from elsewhere, after which it works fine.

**To reproduce**
Set buffer_font_family to "Iosevka".

**Expected behavior**
Either the font changes or nothing happens if it is misconfigured, but it shouldn't become unresponsive.\

**Environment:**
```
Zed 0.45.0 – /Applications/Zed.app 
macOS 12.4 
architecture x86_64
```

[Zed.log](https://github.com/zed-industries/feedback/files/9062247/Zed.log)

**Comment by TheFoolishPupil** [comment]
Hmm, I left the machine for about 30 minutes and it appears to have stabalized. Was unresponsive for at least 10 minutes though.

**Comment by iamnbutler** [comment]
I can't seem to reproduce this one, curious what happened here.

**Comment by logo4poop** [comment]
While I have not changed my font, I am using Yabai myself and it works fine for me.

**Comment by TheFoolishPupil** [comment]
The issue is most definitely related to the font change, as when I just use the default Zed Mono I see no issues either, but thought I would provide the additional context. If you could try to change the font to Iosevka it would be interesting to see if you were able to reproduce the issue? I've downloaded it from [here](https://typeof.net/Iosevka/) and set it in .zed/settings.json with `"buffer_font_family": "Iosevka"`. For me Zed becomes unresponsive the moment I save the settings, and you can easy revery by changing the settings file back from somewhere else.

**Comment by iamnbutler** [comment]
> The issue is most definitely related to the font change, as when I just use the default Zed Mono I see no issues either, but thought I would provide the additional context. If you could try to change the font to Iosevka it would be interesting to see if you were able to reproduce the issue? I've downloaded it from [here](https://typeof.net/Iosevka/) and set it in .zed/settings.json with `"buffer_font_family": "Iosevka"`. For me Zed becomes unresponsive the moment I save the settings, and you can easy revery by changing the settings file back from somewhere else.

Hmm, this is the second time Iosevka has come up. Can you see if this happens with any other font family?

**Comment by TheFoolishPupil** [comment]
@iamnbutler I think the other issue you are referring to was also from me. I will try to see if I can replicate it with another font.

**Comment by mikayla-maki** [comment]
Note: was pairing with @iamnbutler and attempted to set the terminal font to `Iosevka Term`. Nate's Zed crashed and would not open until we removed it from the `~/.zed/setting.json` file. In attempting to reproduce this, I downloaded the 15.6.0 release from [their github](https://github.com/be5invis/Iosevka/releases#gh-light-mode-only) and attempted to install it. It ate all of my computer's application memory and then quit with an error warning that installing it might crash my system. We believe this is because the Iosevka font is some 150mb large when uncompressed, but I did not investigate further. 

I am on a 2020 MacBook Air with 8gb of memory. 
Zed 0.48.1 – /Applications/Zed.app
macOS 12.3.1
architecture arm64

**Comment by jansol** [comment]
I ran into what I suspect is the same issue, while editing the buffer font I wasn't sure of the exact font name for "VictorMono Nerd Font" so went through a couple of variations, and occasionally the buffer would become seemingly read-only? Moving the cursor worked fine, as did (sometimes) backspacing, but writing did not seem to do anything.

After closing and reopening the file everything was working again.

**Comment by winniequinn** [comment]
I can confirm that this still seems to be an issue in Zed 0.77.3 with Iosevka 21.1.1 installed via the [super-ttc-iosevka-21.1.1.zip](https://github.com/be5invis/Iosevka/releases/download/v21.1.1/super-ttc-iosevka-21.1.1.zip) release on an Intel machine running macOS 13.1 (22C65). I believe this may be due to the fact that the "iosevka.ttc" file is 259.6 MB.

**If, however, I install the same font as several smaller files via the [ttc-iosevka-21.1.1.zip](https://github.com/be5invis/Iosevka/releases/download/v21.1.1/ttc-iosevka-21.1.1.zip) release, things mostly work okay.** The editor locks up for 3–5 seconds upon changing the font family setting and upon every launch thereafter—possibly because each of the smaller files is still approximately 40 MB—but it's better than the seemingly permanent hang I run into with the single, massive font file.

For whatever it's worth, I have no issue with either release of Iosevka in Visual Studio Code, IntelliJ, Fleet, Xcode, Terminal, TextEdit, et cetera. I'm not sure what Zed is doing differently here.

(No "zed.log" included because nothing gets logged at all beyond "starting zed" and the setting of environment variables.)

**Comment by JosephTLyons** [comment]
This should be fixed in [v0.119.9-pre](https://zed.dev/releases/preview/0.119.9) - please reopen if the issue persists.
